docs/scenarios_authoring/DataSourcesAndSinks.md

```diff
@@ -34,8 +34,23 @@ The `kafka` sink configuration form will show a list of fields defined in Schema
 
 ![Kafka sink](img/kafkaSink.png "Kafka sink")
 
-If you prefer to define the whole output event as one expression containing value to be written to the Kafka topic, instead of filling separete autogenerated fields, you can do that by switching `Raw editor` to `true`.
+If you prefer to define the whole output event as one expression containing value to be written to the Kafka topic, instead of filling separate autogenerated fields, you can do that by switching `Raw editor` to `true`.
 
 When using the Raw editor, you can choose the validation mode:
-* strict - requires providing all fields, including optional fields, without redundant fields;
-* lax - requires providing only required fields, you can skip optional fields and add redundant fields.
\ No newline at end of file
+* strict - requires providing all fields, including optional fields, without redundant fields, requires exact matching;
+* lax - requires providing only required fields, you can skip optional fields and add redundant fields, exact matching is not required;
+
+(*) allowing/denying redundant fields works only in Avro schema. In JSON schema this configuration is provided
+explicitly by schema property: `additionalProperties`. See more about how to use [JSON Schema](/docs/scenarios_authoring/HandlingSchemas.md#json-schema).
+
+(*) exact matching describes how output validation works for types Unknown and Union. Sometimes helpers or enrichers can
+return data with type `Unknown` - it means we can't recognize the type, consequently, we have a situation difficult to handle.
+And it's the place where exact matching comes to help us:
+
+* strict mode doesn't allow to pass Unknown for any schema type. In a situation when we try to pass a value with type
+  union as the output with schema union, strict mode requires these two unions should be the same;
+
+* lax mode allows passing Unknown for any schema type. In the case of unions, lax mode requires a situation when just only one type
+  of value union matches to schema union. But be aware of it, because when data will not match the output schema
```

Comment:
I do not understand the sentence starting with "but" in the context of the previous sentence. The reason for this may be my little knowledge of union type. One way or another we need a link to an external document on this union thing, because majority of readers may have problem understanding this part.

Reply:
but not only Unknown. Just wider type. 
Eg. Schema requires `Map[String,String]`, and you can pass `Map[String, Any]`

Reply:
As I know and understand `Any` is typed to `Unknown`?